### PR TITLE
RATIS-1768. Fix stepDown command don't work issue

### PR DESCRIPTION
**Pull request by codings-dan**
see https://issues.apache.org/jira/browse/RATIS-1768

### CODE REVIEW

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java [hunk 1]
```diff
@@ -1177,7 +1177,8 @@ CompletableFuture<RaftClientReply> stepDownLeaderAsync(TransferLeadershipRequest
     assertGroup(request.getRequestorId(), request.getRaftGroupId());
 
     return role.getLeaderState().map(leader -> leader.submitStepDownRequestAsync(request))
-        .orElseGet(() -> CompletableFuture.completedFuture(newSuccessReply(request)));
+        .orElseGet(() -> CompletableFuture.completedFuture(newExceptionReply(
+            request, new NotLeaderException(getMemberId(), getRaftConf().getPeer(info.getLeaderId()), null))));
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java:1181]: Use `generateNotLeaderException()`, i.e.
```java
        .orElseGet(() -> CompletableFuture.completedFuture(newExceptionReply(request, generateNotLeaderException());
```

**Review by szetszwo — COMMENTED** [review]
@codings-dan , thanks a lot for working on this!  Just a minor comment inlined.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java (hunk 1 above)
Comment (codings-dan) [reply ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java]: done

**Comment by codings-dan** [comment]
@szetszwo Thanks for reviewing the pull request, PTAL

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**